src/python/pants/backend/adhoc/target_types.py

```diff
@@ -297,6 +299,58 @@ def enum_value(self) -> PathEnvModifyMode:
         return PathEnvModifyMode(self.value)
 
 
+class AdhocToolOutputsMatchErrorBehavior(StringField):
+    alias = "outputs_match_error_behavior"
+    default = GlobMatchErrorBehavior.error.value
```

Comment:
This is a new behaviour, so I think we should introduce it as:

```suggestion
    default = GlobMatchErrorBehavior.warn.value
```

To avoid it being a hard-break that makes it harder for people to upgrade (especially to quickly try a pre-release version).

Reply:
Makes sense. Will make the edit now subject to my comment about an alternate developer experience (and where the default in that scheme would be `all_warn`).